oi/ContainerInfo.h

```diff
@@ -78,6 +85,10 @@ struct ContainerInfo {
   bool operator<(const ContainerInfo& rhs) const {
     return (typeName < rhs.typeName);
   }
+
+  std::regex getMatcher() const {
+    return std::regex("^" + typeName + "<|^" + typeName + "$");
```

Comment:
It probably makes more sense to keep this as a private field and return a reference, as I understand it this constructor parses the regex so is probably quite expensive.